thewalrus/_hafnian.py

```diff
@@ -55,6 +57,32 @@ def input_validation(A, rtol=1e-05, atol=1e-08):
     return True
 
 
+def bandwidth(A):
+    """Calculates the upper bandwidth of the matrix A.
+    Args:
+        A (array): input matrix
+    Returns:
+        (int): bandwidth of matrix
+    """
+    n, _ = A.shape
+    for i in range(n):
+        vali = np.diag(A, i)
+        if np.allclose(vali, 0):
+            return i - 1
+    return n - 1
+
+
+def powerset(iterable):
+    """Calculates the powerset of a list.
+    Args:
+        iterable (list): input list
```

Comment:
if you require the input to be a list then there's no need to cast it below. Or if you can remove the casting, you can say `iterable (iterable): ...`